Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+namespace BootCamp.Chapter
+{
+    public static class Lesson4
+    {
+        public static void Demo()
+        {
+            char again = 'y';
+
+            while (again == 'y')
+            {
+                string firstName = RequestString("Please enter first name:");
+
+                string surname = RequestString("Please enter last name:");
+
+                int age = RequestInt("Please enter age in years:");
+
+                float weightInKilograms = RequestFloat("Please enter weight in kilograms:");
+
+                float heightInMeters = RequestFloat("Please enter height in meters:");
+
+                ReportBasicInfo(firstName, surname, age, weightInKilograms, heightInMeters);
+
+                ReportBodyMassIndex(firstName, weightInKilograms, heightInMeters);
+
+                again = RequestUserChoiceToRepeatOrNot();
+            }
+        }
+
+        public static string RequestString(string prompt)
+        {
+            Console.Write($"{prompt}{Environment.NewLine}");
+            return ValidStringOrErrorCode();
+        }
+
+        private static string ValidStringOrErrorCode()
+        {
+            var userInput = Console.ReadLine();
+            if (NullStringCheck(userInput))
+            {
+                Console.Write($"Name cannot be empty.");
+                return "-";
+            }
+
+            return userInput;
+        }
+
+        public static int RequestInt(string prompt)
+        {
+            Console.Write($"{prompt}{Environment.NewLine}");
+            var userInput = Console.ReadLine();
+            if (NullStringCheck(userInput))
+            {
+                return 0;
+            }
+            return ValidIntOrErrorCode(userInput);
+        }
+
+        static int ValidIntOrErrorCode(string input)
+        {
+           
+            var isInt = int.TryParse(input, out var number);
+            if (!isInt)
+            {
+                Console.Write($"\"{input}\" is not a valid number.");
+                return -1;
+            }
+
+            return number;
+        }
+
+        public static float RequestFloat(string prompt)
+        {
+            Console.Write($"{prompt}{Environment.NewLine}");
+            var userInput = Console.ReadLine();
+            if (NullStringCheck(userInput))
+            {
+                Console.Write("");
+                return 0;
+            }
+            return ValidFloatOrErrorCode(userInput);
+        }
+
+        static float ValidFloatOrErrorCode(string input)
+        {
+            var isFloat = float.TryParse(input, out var number);
+            if (!isFloat)
+            {
+                Console.Write($"\"{input}\" is not a valid number.");
+                return -1;
+            }
+            else
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'else' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884596)

Reply:
removed else